maestrowf/datastructures/yamlspecification.py

```diff
@@ -144,12 +159,17 @@ def load_specification_from_stream(cls, stream):
 
     def verify(self):
         """Verify the whole specification."""
-        self.verify_description()
-        self.verify_study()
-        self.verify_parameters()
+        try:
+            self.verify_description()
+            self.verify_study()
+            self.verify_parameters()
+        except jsonschema.ValidationError as e:
+            logger.error(e.message)
+            sys.exit(1)
```

Comment:
I'm unsure how I feel about have an `sys.exit` here -- maybe it's because I was always taught that the number of exit points should be as minimal as possible.  I noticed that we aren't catching the exception in `maestro.py`, I'm almost wondering if this try catch should be wrapping when we load the study. What do you think @ben-bay?

Reply:
I think I catch your drift. I just pushed up my response to your comments.